server/src/test/java/org/elasticsearch/ExceptionSerializationTests.java

```diff
@@ -888,16 +887,16 @@ public void testElasticsearchRemoteException() throws IOException {
     public void testShardLockObtainFailedException() throws IOException {
         ShardId shardId = new ShardId("foo", "_na_", 1);
         ShardLockObtainFailedException orig = new ShardLockObtainFailedException(shardId, "boom");
-        Version version = VersionUtils.randomIndexCompatibleVersion(random());
-        ShardLockObtainFailedException ex = serialize(orig, version.transportVersion);
+        TransportVersion version = TransportVersionUtils.randomCompatibleVersion(random());
```

Comment:
I'm not sure why this was using index compatible version, this should only be used in serialization between nodes

Reply:
agree, it looks like it is not related to index compatibility